clusterloader2/pkg/measurement/common/metrics_for_e2e.go

```diff
@@ -0,0 +1,149 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/test/e2e/framework/metrics"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	metricsForE2EName     = "MetricsForE2E"
+	caFunctionMetric      = "cluster_autoscaler_function_duration_seconds_bucket"
+	caFunctionMetricLabel = "function"
+)
+
+var interestingApiServerMetricsLabels = []string{
+	"apiserver_request_count",
+	"apiserver_request_latencies_summary",
+	"etcd_helper_cache_entry_count",
+	"etcd_helper_cache_hit_count",
+	"etcd_helper_cache_miss_count",
+	"etcd_request_cache_add_latencies_summary",
+	"etcd_request_cache_get_latencies_summary",
+	"etcd_request_latencies_summary",
+}
+
+var interestingControllerManagerMetricsLabels = []string{
+	"garbage_collector_attempt_to_delete_queue_latency",
+	"garbage_collector_attempt_to_delete_work_duration",
+	"garbage_collector_attempt_to_orphan_queue_latency",
+	"garbage_collector_attempt_to_orphan_work_duration",
+	"garbage_collector_dirty_processing_latency_microseconds",
+	"garbage_collector_event_processing_latency_microseconds",
+	"garbage_collector_graph_changes_queue_latency",
+	"garbage_collector_graph_changes_work_duration",
+	"garbage_collector_orphan_processing_latency_microseconds",
+
+	"namespace_queue_latency",
+	"namespace_queue_latency_sum",
+	"namespace_queue_latency_count",
+	"namespace_retries",
+	"namespace_work_duration",
+	"namespace_work_duration_sum",
+	"namespace_work_duration_count",
+}
+
+var interestingKubeletMetricsLabels = []string{
+	"kubelet_container_manager_latency_microseconds",
+	"kubelet_docker_errors",
+	"kubelet_docker_operations_latency_microseconds",
+	"kubelet_generate_pod_status_latency_microseconds",
+	"kubelet_pod_start_latency_microseconds",
+	"kubelet_pod_worker_latency_microseconds",
+	"kubelet_pod_worker_start_latency_microseconds",
+	"kubelet_sync_pods_latency_microseconds",
+}
+
+func init() {
+	measurement.Register(metricsForE2EName, createmetricsForE2EMeasurement)
+}
+
+func createmetricsForE2EMeasurement() measurement.Measurement {
+	return &metricsForE2EMeasurement{}
+}
+
+type metricsForE2EMeasurement struct {
+	clusterAutoscalerMetricsBeforeTest metrics.MetricsCollection
+}
+
+// Execute gathers and prints e2e metrics data.
+func (m *metricsForE2EMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+
+	provider, err := util.GetString(config.Params, "provider")
+	if err != nil {
+		return summaries, err
+	}
+
+	grabMetricsFromKubelets, err := util.GetBoolOrDefault(config.Params, "gatherKubletsMetrics", false)
```

Comment:
Where are you adding this new flag? Also please correct Kubelet spelling.

Reply:
Type fixed.

Reply:
In original tests, kubelets metric were gathered based on number of nodes. I want to have the same logic here. However, IMHO measurement shouldn't decide what will be gather, this should be done on test definition level. That's why I add flag for this functionality - to make it settable from test level.

Reply:
Ok, that sounds reasonable to me.